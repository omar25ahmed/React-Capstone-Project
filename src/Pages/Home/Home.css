.stats {
  background-color: #35558a;
}

p {
  margin: 0;
}

.break-div {
  padding: 0;
}

@media only screen and (min-width: 768px) {
  .stats {
    width: 50%;
    padding: 10px;
  }
}
