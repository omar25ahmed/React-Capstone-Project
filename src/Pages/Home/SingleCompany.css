.company {
  list-style: none;
  cursor: pointer;
  height: 150px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  transition: all 0.4s;
}

li {
  background-color: #4066ae;
}

li:nth-child(3n - 2) {
  background-color: #4063a5;
}

.symbol {
  color: #2d4472;
  font-size: 50px;
  font-style: italic;
}

.title {
  font-weight: bold;
  font-size: 10px;
}

.company:hover {
  background-color: #164fb9;
}
