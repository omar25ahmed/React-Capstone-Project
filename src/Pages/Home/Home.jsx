/* eslint-disable no-nested-ternary */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { FaSearch } from 'react-icons/fa';
import { fetchCompanies } from '../../Redux/companiesReducer';
import SingleCompany from './SingleCompany';
import HeadingComponents from '../HeadingComponent';
import './Home.css';

function Home() {
  const dispatch = useDispatch();
  const companies = useSelector((state) => state);
  const [searchTerm, setSearchTerm] = useState('');

  useEffect(() => {
    dispatch(fetchCompanies());
  }, []);

  return (
    <div data-testid="companies">
      <HeadingComponents name="Top Companies" title="in Canada" />
      <div style={{ padding: '0' }} className="d-md-flex container">
        <div className="stats">
          <p>MOST IMPORTANT COMPANIES</p>
        </div>
        <div className="input-group">
          <div className="input-group-prepend d-md-none">
            <span className="input-group-text bg-primary" id="basic-addon1"><FaSearch size="1.5rem" color="#fff" /></span>
          </div>
          <input type="text" className="form-control" placeholder="Search Companies..." aria-label="Username" aria-describedby="basic-addon1" onChange={(e) => setSearchTerm(e.target.value)} />
        </div>
      </div>
      <div className="container">
        <div className="row">
          {companies
            .filter((company) => (searchTerm === ''
              ? company
              : company.companyName.toLowerCase().includes(searchTerm.toLowerCase())
                ? company
                : ''))
            .map((company) => (
              <SingleCompany key={company.volume} company={company} />
            ))}
        </div>
      </div>
    </div>
  );
}

export default Home;
