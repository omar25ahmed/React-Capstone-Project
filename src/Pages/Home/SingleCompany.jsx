import React from 'react';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import './SingleCompany.css';
import { FaArrowAltCircleRight } from 'react-icons/fa';

function SingleCompany({ company }) {
  const { companyName, symbol } = company;
  const navigate = useNavigate();
  const openCompanyProfileHandler = () => {
    navigate(`/${companyName}`);
  };
  return (
    <li data-testid="company" aria-hidden="true" onClick={openCompanyProfileHandler} className="col-6 col-md-3 company">
      <div className="symbol">{symbol}</div>
      <div className="title">{companyName}</div>
      <div className="icon"><FaArrowAltCircleRight /></div>
    </li>
  );
}

export default SingleCompany;

SingleCompany.defaultProps = {
  company: {
    companyName: '',
    symbol: '',
  },
};

SingleCompany.propTypes = {
  company: PropTypes.shape({
    companyName: PropTypes.string,
    symbol: PropTypes.string,
  }),
};
