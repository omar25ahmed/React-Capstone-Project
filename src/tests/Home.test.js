import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Redux/Store';
import Home from '../Pages/Home/Home';

describe('testing Home component', () => {
  it('snapshot for the Home ', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Home component renders correctly', () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(screen.getByTestId('companies')).toBeInTheDocument();
  });
});
