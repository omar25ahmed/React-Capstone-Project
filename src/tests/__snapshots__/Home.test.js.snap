// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home component snapshot for the Home  1`] = `
<div
  data-testid="companies"
>
  <section>
    <div
      className="container d-flex align-items-center justify-content-center p-3"
    >
      <div
        className="me-2"
      >
        <img
          alt="canada flag"
          className=""
          src="Canada1.png"
        />
      </div>
      <div>
        <h1
          className="display-4"
        >
          Top Companies
        </h1>
        <small
          className="ms-4"
        >
          in Canada
        </small>
      </div>
    </div>
  </section>
  <div
    className="d-md-flex container"
    style={
      Object {
        "padding": "0",
      }
    }
  >
    <div
      className="stats"
    >
      <p>
        MOST IMPORTANT COMPANIES
      </p>
    </div>
    <div
      className="input-group"
    >
      <div
        className="input-group-prepend d-md-none"
      >
        <span
          className="input-group-text bg-primary"
          id="basic-addon1"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="1.5rem"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#fff",
              }
            }
            viewBox="0 0 512 512"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </span>
      </div>
      <input
        aria-describedby="basic-addon1"
        aria-label="Username"
        className="form-control"
        onChange={[Function]}
        placeholder="Search Companies..."
        type="text"
      />
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
